src/repart/repart.c

```diff
@@ -7821,6 +7867,73 @@ static int parse_partition_types(const char *p, GptPartitionType **partitions, s
         return 0;
 }
 
+static int parse_join_signature(const char *p, Hashmap **verity_settings_map) {
+        _cleanup_(verity_settings_freep) VeritySettings *verity_settings = NULL;
+        _cleanup_free_ char *root_hash = NULL, *base = NULL, *signature = NULL;
+        _cleanup_free_ void *content = NULL;
+        size_t len;
+        int r;
+
+        assert(p);
+        assert(verity_settings_map);
+
+        r = extract_many_words(&p, ":", 0, &root_hash, &base, &signature);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse signature parameter '%s': %m", p);
+        if (r < 2 || r > 3)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Expected hash:sig");
```

Comment:
this will parse things quite differently from --root-hash= on the nspawn and elsewhere, because `--root-hash=file:foo` will load from a file `file:foo` there, but here we'd refuse `--join-signature=943759218759827:file:foo`...

I think we should expose similar behaviour. or in other words, change extract-many_words() to only extract two words, not three, and then use startswith() to look for "base64"

Reply:
--root-hash in nspawn does not support `file:` ? It's always inline hex as far as I can tell. It's the signature that can be a file

Reply:
also extract_many_words only returns `1234` and `base64` with 2 parameters, the third one is left in the input string

Reply:
the "file" was just some example string, it carried no meaning